### PR TITLE
TD-998-Passporting: Issues on 'Learning portal' due to Passporting changes

### PR DESCRIPTION
**Pull request by Auldrin-Possa**
**JIRA link**

**Description**
Implemented passporting changes.

**Screenshots**
Updated in Jira ticket.

-----
**Developer checks**

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin]
- [ ] Updated/added documentation in [Confluence]
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EnrolController.cs [hunk 1]
```diff
@@ -255,17 +255,15 @@ public IActionResult EnrolDelegateSummary(int delegateId, string delegateName)
...
             var clockUtility = new ClockUtility();
             var sessionEnrol = multiPageFormService.GetMultiPageFormData<SessionEnrolDelegate>(
                MultiPageFormDataFeature.EnrolDelegateInActivity,
                TempData);
             if (!sessionEnrol.IsSelfAssessment)
             {
-                progressDataService.CreateNewDelegateProgress(delegateId, sessionEnrol.AssessmentID.GetValueOrDefault(), sessionEnrol.AssessmentVersion,
-                  clockUtility.UtcNow, 0, GetAdminID(), sessionEnrol.CompleteByDate, sessionEnrol.SupervisorID.GetValueOrDefault());
-
                 enrolService.EnrolDelegateOnCourse(delegateId, sessionEnrol.AssessmentID.GetValueOrDefault(), sessionEnrol.AssessmentVersion, 0, GetAdminID(), sessionEnrol.CompleteByDate, sessionEnrol.SupervisorID.GetValueOrDefault(), "AdminEnrolDelegateOnCourse");
```
Comment (martinfoster-hee) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EnrolController.cs:266]: Was this the duplicate call? Just curious! (Deleting existing code makes me anxious lol).

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
I think the changes to return to the Completed activities view all need reverting @Auldrin-Possa , unless they do something I don't understand?

File: DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs [hunk 2]
```diff
@@ -158,7 +158,7 @@ public IActionResult Close()
         {
             sessionService.StopDelegateSession(User.GetCandidateIdKnownNotNull(), HttpContext.Session);
 
-            return RedirectToAction("Current", "LearningPortal");
+            return RedirectToAction("Completed", "LearningPortal");
```
[the remark this answers is not in the document]
Comment (Auldrin-Possa) [reply DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs:161]: links are used in the completed section and working correctly.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EnrolController.cs [hunk 3]
```diff
@@ -255,17 +255,15 @@ public IActionResult EnrolDelegateSummary(int delegateId, string delegateName)
...
         {
+            var delegateUserId = User.GetUserIdKnownNotNull();
+            var centreId = User.GetCentreIdKnownNotNull();
             var clockUtility = new ClockUtility();
             var sessionEnrol = multiPageFormService.GetMultiPageFormData<SessionEnrolDelegate>(
                MultiPageFormDataFeature.EnrolDelegateInActivity,
                TempData);
             if (!sessionEnrol.IsSelfAssessment)
             {
-                progressDataService.CreateNewDelegateProgress(delegateId, sessionEnrol.AssessmentID.GetValueOrDefault(), sessionEnrol.AssessmentVersion,
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EnrolController.cs:264]: This code is needed. Please revert this change. We will need to set up and poulate delegateId variable above to pass into it, too.

**Comment by Auldrin-Possa** [comment]
> 



> I think the changes to return to the Completed activities view all need reverting @Auldrin-Possa , unless they do something I don't understand?

Reverted.

**Comment by kevwhitt-hee** [comment]
Morning @Auldrin-Possa, there is a build fail on this branch now. Please could you look into fixing it?

**Comment by Auldrin-Possa** [comment]
> Morning @Auldrin-Possa, there is a build fail on this branch now. Please could you look into fixing it?

Resolved.

**Review by kevwhitt-hee — APPROVED** [review]

File: DigitalLearningSolutions.Data/DataServices/CourseDataService.cs [hunk 4]
```diff
@@ -426,7 +430,7 @@ FROM Candidates
             {
                 logger.LogWarning(
                     "Not enrolled delegate on self assessment as db insert failed. " +
-                    $"Self assessment id: {selfAssessmentId}, candidate id: {candidateId}"
+                    $"Self assessment id: {selfAssessmentId}, user id: {delegateUserId}"
                 );
```
Comment (martinfoster-hee) [inline DigitalLearningSolutions.Data/DataServices/CourseDataService.cs:434]: Super-pedantic but personally I'd have put this as 'delegate user id' so future debuggers know we aren't talking about say an admin user id here. :)

**Review by martinfoster-hee — COMMENTED** [review]
Looks good to me.

**Review by kevwhitt-hee — APPROVED** [review]

**Pull request merged**